tests/blackbox_test.go

```diff
@@ -143,6 +151,46 @@ func outer(t *testing.T, test types.Test, negativeTests bool) error {
 		}
 		test.Out[i].SetOffsets()
 
+		// Replace all UUID variables (format $uuid<num>) in configs and partitions with UUID
+		test.Config, err = replaceUUIDVars(t, test.Config, UUIDmap)
+		if err != nil {
+			return err
+		}
+
+		for _, disk := range test.In {
+			for _, partition := range disk.Partitions {
+				partition.TypeGUID, err = replaceUUIDVars(t, partition.TypeGUID, UUIDmap)
```

Comment:
Why are we running replace against `TypeGUID`?

Reply:
Why not? As far as Ignition is concerned its just another thing that can match or not match.

Reply:
My understanding of the original driving force that caused the initial discussion that led to this PR was that we wanted to eliminate duplicate UUIDs in fields where having non-unique entries across multiple tests could lead to parallelism issues.

I'm not particularly against doing this but I don't think it really provides much if any value to do so.

Reply:
It's also to allow you to say "parition N should look like this, fix it if it doesnt" and type guids are just as much a part of that. Ignoring what this might be used for, its still a code path in Ignition that should get testing.

Reply:
The code path is still being tested regardless of if the UUID is generated or not.

I will concede that it doesn't particularly matter what the `TypeGUID` is in most cases though and for ease of writing having the functionality applied makes sense.

Reply:
Oh, yeah it doesn't really matter. The $uuidN is just much easier to read/write. If we really wanted to mock out real world use cases we could include all of the special type guid (e.g. coreos root on raid) in the templating function.